sql/catalog: improve redaction of privilege validation errors

spilchen opened the pull request:
We had a sentry error come in for a privilege validation error. The error looked like this:
```
privilege.go:397: relation × (52): ×
```

This change improves things so not everything is redacted. It now looks something like this:
```
privilege.go:397: relation × (52): user root must have exactly [ALL] privileges on table ‹×›
```

Closes #152912
Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153798)
<!-- Reviewable:end -->

rafiss reviewed (approved):
nice find!

spilchen committed:
sql/catalog: improve redaction of privilege validation errors

We had a sentry error come in for a privilege validation error. The
error looked like this:
```
privilege.go:397: relation × (52): ×
```

This change improves things so not everything is redacted. It now looks
something like this:
```
privilege.go:397: relation × (52): user root must have exactly [ALL] privileges on table ‹×›
```

Closes #152912
Epic: none
Release note: none
